<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FormDeviserPackage</class>
 <widget class="QWidget" name="FormDeviserPackage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>749</width>
    <height>448</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QWidget" name="widget" native="true">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>120</height>
      </size>
     </property>
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span&gt;The short &lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;name&lt;/span&gt;&lt;span&gt; or &lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;label&lt;/span&gt;&lt;span&gt; given to this package. This will be used in the associated URL and as the package XML prefix.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Name:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="txtName">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The short &lt;span style=&quot; font-weight:600;&quot;&gt;name&lt;/span&gt; or &lt;span style=&quot; font-weight:600;&quot;&gt;label&lt;/span&gt; given to this package. This will be used in the associated URL and as the package XML prefix. This field is required.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QLineEdit { background-color: rgba(255,0,0,127);selection-background-color:rgba(255,0,0,127) }QLineEdit:hover { background-color: rgba(255,0,0,127);selection-background-color:rgba(255,0,0,127) }</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="minimumSize">
         <size>
          <width>80</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Full Name: &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="txtFullName">
        <property name="toolTip">
         <string>The full title of the package as it would be written in a specification document. This field is required.</string>
        </property>
        <property name="styleSheet">
         <string notr="true">QLineEdit { background-color: rgba(255,0,0,127);selection-background-color:rgba(255,0,0,127) }QLineEdit:hover { background-color: rgba(255,0,0,127);selection-background-color:rgba(255,0,0,127) }</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="chkRequired">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span&gt;All SBML Level 3 packages must state whether the package may change the mathematical interpretation of the core model. This is the value given to the &lt;/span&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;required&lt;/span&gt;&lt;span&gt; attribute on the &amp;lt;&lt;/span&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;sbml&amp;gt;&lt;/span&gt;&lt;span&gt; element. Check this box if this package is required.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>required</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="chkRequiresAdditionalCode">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span&gt;In some cases it may be necessary to add code that is very specific to a package and will not be generated by deviser.  Check this box if you wish to specify the location of such code to be used in the generation of the code for this class.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>requires additional code</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;justify&quot;&gt;&lt;br/&gt;A package must define the starting point for the enumeration of SBMLTypeCode_t values used to distinguish SBase derived objects. This should be an integer multiple of 100. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="widget_2" native="true">
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="1">
       <widget class="QLineEdit" name="txtNumber">
        <property name="styleSheet">
         <string notr="true">QLineEdit { background-color: rgba(255,0,0,127);selection-background-color:rgba(255,0,0,127) }QLineEdit:hover { background-color: rgba(255,0,0,127);selection-background-color:rgba(255,0,0,127) }</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="minimumSize">
         <size>
          <width>85</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Number: &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QToolButton" name="cmdShowNumbers">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../deviser-resources.qrc">
          <normaloff>:/images/dialog-information.png</normaloff>:/images/dialog-information.png</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;justify&quot;&gt;&lt;br/&gt;A package must define the additional offset added to all validation rule numbers to allow these to be uniquely stored. This should be a two digit integer followed by 00000 for example 1900000. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="textFormat">
      <enum>Qt::RichText</enum>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="widget_3" native="true">
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="minimumSize">
         <size>
          <width>85</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Offset: &lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="txtOffset">
        <property name="styleSheet">
         <string notr="true">QLineEdit { background-color: rgba(255,0,0,127);selection-background-color:rgba(255,0,0,127) }QLineEdit:hover { background-color: rgba(255,0,0,127);selection-background-color:rgba(255,0,0,127) }</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QToolButton" name="cmdShowOffset">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset resource="../deviser-resources.qrc">
          <normaloff>:/images/help_about.png</normaloff>:/images/help_about.png</iconset>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>350</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string> Additional code files: </string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Declaration: </string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLineEdit" name="txtDeclaration">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span&gt;The location of an additional &lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;header&lt;/span&gt;&lt;span&gt; file that should be included within the generated code.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="cmdBrowseDeclaration">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Implementation: </string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLineEdit" name="txtImplementation">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span&gt;The location of an additional &lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;code&lt;/span&gt;&lt;span&gt; file that should be included within the generated code.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="cmdBrowseImplementation">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Custom Copyright: </string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_21">
        <item>
         <widget class="QLineEdit" name="txtCopyright">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The location of an optionall &lt;span style=&quot; font-weight:600;&quot;&gt;custom copyright&lt;/span&gt; file that should be included within the generated code.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="cmdBrowseCopyright">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../deviser-resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>txtDeclaration</sender>
   <signal>textEdited(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>declarationChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>411</x>
     <y>377</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtFullName</sender>
   <signal>textEdited(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>fullNameChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>63</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtImplementation</sender>
   <signal>textEdited(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>implementationChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>411</x>
     <y>409</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtName</sender>
   <signal>textEdited(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>nameChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>33</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtNumber</sender>
   <signal>textEdited(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>numberChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>188</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtOffset</sender>
   <signal>textEdited(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>offsetChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkRequired</sender>
   <signal>stateChanged(int)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>requiredStateChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>423</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chkRequiresAdditionalCode</sender>
   <signal>stateChanged(int)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>additionalCodeStateChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>423</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseDeclaration</sender>
   <signal>clicked()</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>browseDeclaration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>707</x>
     <y>377</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseImplementation</sender>
   <signal>clicked()</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>browseImplementation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>707</x>
     <y>409</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdShowNumbers</sender>
   <signal>clicked()</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>showNumbersAndOffsets()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>719</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdShowOffset</sender>
   <signal>clicked()</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>showNumbersAndOffsets()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>719</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtName</sender>
   <signal>textChanged(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>nameModified(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>417</x>
     <y>29</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtFullName</sender>
   <signal>textChanged(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>fullNameModified(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>417</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtOffset</sender>
   <signal>textChanged(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>offsetModified(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>405</x>
     <y>290</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtNumber</sender>
   <signal>textChanged(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>numberModified(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>405</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmdBrowseCopyright</sender>
   <signal>clicked()</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>browseCopyright()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>716</x>
     <y>417</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>txtCopyright</sender>
   <signal>textEdited(QString)</signal>
   <receiver>FormDeviserPackage</receiver>
   <slot>copyrightChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>417</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>nameChanged(QString)</slot>
  <slot>fullNameChanged(QString)</slot>
  <slot>requiredStateChanged(int)</slot>
  <slot>additionalCodeStateChanged(int)</slot>
  <slot>numberChanged(QString)</slot>
  <slot>offsetChanged(QString)</slot>
  <slot>declarationChanged(QString)</slot>
  <slot>implementationChanged(QString)</slot>
  <slot>browseDeclaration()</slot>
  <slot>browseImplementation()</slot>
  <slot>showNumbersAndOffsets()</slot>
  <slot>nameModified(QString)</slot>
  <slot>fullNameModified(QString)</slot>
  <slot>numberModified(QString)</slot>
  <slot>offsetModified(QString)</slot>
  <slot>browseCopyright()</slot>
  <slot>copyrightChanged(QString)</slot>
 </slots>
</ui>
